online-shop/src/app/store/reducers/cart.reducers.ts

```diff
@@ -0,0 +1,73 @@
+import { createReducer, on } from "@ngrx/store";
+import { initialState } from "../state/product.state";
+import {
+  addToCart,
+  addToCartError,
+  addToCartSuccess,
+  checkout,
+  checkoutError,
+  checkoutSuccess
+} from "../actions/cart.actions";
+import {getProduct, getProductError, getProductSuccess} from "../actions/product.actions";
+import {initialCartState} from "../state/cart.state";
+
+export const cartReducer = createReducer(
+  initialCartState,
+
+  on(addToCart, (state, {id}) => {
+    let index = state.products.find(products => products.productId === id);
```

Comment:
Here I think would be more clear to use product instead of index. Name "index" is more used for numbers, like counters for ex